_source/_posts/2020-08-05-java-arm-performance.md

```diff
@@ -0,0 +1,112 @@
+
+In this post, I'll share the Java benchmarks I took on various AWS EC2 instances, and, just for fun, on my laptop. 
+* Amazon a1.large (ARMv8 Cortex-A72, 2 Cores, 4GB RAM)
+* Amazon m6g.medium (ARMv8 Neoverse-N1, 1 Core, 4GB RAM)
+* Amazon t3.medium (Intel Xeon Platinum 8259CL, 1 Core / 2 Threads, 4GB RAM)
+* Apple MacBook Pro (Intel i9 2.4GHz, 8 Core / 16 Threads, 64GB RAM)
+
+## A Note About Benchmarks
+
+Benchmarks are just numbers. They serve as a starting point when you are figuring out the compute power you need for your own application. All applications are different, your workload will likely have different characteristics than these benchmarks. The only way to figure out how your application will perform on a different system is to try to test it out!
+
+For these tests, I tried to compare three different AWS offerings that are similar and have comparable on-demand pricing. There are some differences though. The a1.large instance is from Amazon's first-generation ARM processors, whereas the m6g.medium is the current Arm series, and the t3.medium is an Intel x86_64 processor.
+
+To keep things consistent, all of these benchmarks used Amazon Corretto 11.0.7.10.1 JVM, with the default GC configuration, and with the tests run through [Phoronix Test Suite](https://www.phoronix-test-suite.com/).
+
+## The Benchmarks!
+
+In almost all cases the a1.large instance performed the worst, and my MacBook the best. This isn't particularly interesting, so I'm going to focus my analysis on the differences between the t3.medium and the m6g.medium instances.
+
+## Machine Learning
+
+First up, we have a test based on Apache Spark's MLlib which uses a random forest algorithm.
+
+{% img blog/java-arm-performance/apache-spark-random-forest.png alt:"Graph showing the t3 is faster" width:"800" %}
+
+In this test, the t3.medium is 15% faster than the m6g.medium.
+
+{% img blog/java-arm-performance/apache-spark-als.png alt:"Graph showing the Arm server performed better" width:"800" %}
+
+The Spark alternating least squares (ALS) benchmark is one of the few tests where both the Arm servers outpaced the t3.medium.
+
+{% img blog/java-arm-performance/apache-spark-bayes.png alt:"Graph showing the m6g was faster" width:"800" %}
+
+In the Spark Naive Bayes algorithm test, the m6g.medium was 8% faster than the t3.
+
+**Winner:** m6g.medium. this was almost too close to call, but in the words of Meat Loaf,  "[Two Out of Three Ain't Bad.](https://youtu.be/k5hWWe-ts2s)"
```

Comment:
I think you have to embed this video! :) That way people don't leave the page and they can listen to music while they finish reading. Brilliant!